apps/web/pages/api/cancel.ts

```diff
@@ -197,21 +198,52 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     bookingToDelete.user.credentials.push(FAKE_DAILY_CREDENTIAL);
   }
 
-  const apiDeletes = async.mapLimit(bookingToDelete.user.credentials, 5, async (credential: Credential) => {
-    const bookingRefUid = bookingToDelete.references.filter((ref) => ref.type === credential.type)[0]?.uid;
-    const bookingExternalCalendarId = bookingToDelete.references.filter(
-      (ref) => ref.type === credential.type
-    )[0]?.externalCalendarId;
-    if (bookingRefUid) {
-      if (credential.type.endsWith("_calendar")) {
-        const calendar = getCalendar(credential);
+  const apiDeletes = [];
 
-        return calendar?.deleteEvent(bookingRefUid, evt, bookingExternalCalendarId);
-      } else if (credential.type.endsWith("_video")) {
-        return deleteMeeting(credential, bookingRefUid);
+  const calendarReference = bookingToDelete.references.find((reference) =>
+    reference.type.includes("_calendar")
+  );
+
+  if (calendarReference) {
+    const { credentialId, uid, externalCalendarId } = calendarReference;
+    if (credentialId) {
+      const calendarCredential = bookingToDelete.user.credentials.find(
+        (credential) => credential.id === credentialId
+      );
+      if (calendarCredential) {
+        const calendar = getCalendar(calendarCredential);
+        apiDeletes.push(calendar?.deleteEvent(uid, evt, externalCalendarId));
       }
+    } else {
+      bookingToDelete.user.credentials
+        .filter((credential) => credential.type.endsWith("_calendar"))
+        .forEach((credential) => {
+          const calendar = getCalendar(credential);
+          apiDeletes.push(calendar?.deleteEvent(uid, evt, externalCalendarId));
+        });
     }
```

Comment:
The thing I'm noticing here is that it is not very clear or legible what is happening here. Any reviewer and future dev coming to this code should understand a few things.
- What is this code doing?
- Why is this code needed?
- Why is this code the way it is today?

I kinda understand what is happening here but I have to spend like 15 minutes to gather all the pieces and understand what is happening here.

We should prioritize legibility above all since we're not working alone. We can improve this with clearer variable names explain the WHAT and good comments to explain the WHY.